services/src/main/java/com/linkedin/kmf/common/Utils.java

```diff
@@ -35,14 +40,54 @@
    * @return the number of partitions of the given topic
    */
   public static int getPartitionNumForTopic(String zkUrl, String topic) {
-    ZkUtils zkUtils = ZkUtils.apply(zkUrl, 30000, 30000, JaasUtils.isZkSecurityEnabled());
+    ZkUtils zkUtils = ZkUtils.apply(zkUrl, ZK_SESSION_TIMEOUT, ZK_CONNECTION_TIMEOUT, JaasUtils.isZkSecurityEnabled());
     Seq<String> topics = scala.collection.JavaConversions.asScalaBuffer(Arrays.asList(topic));
     int partition = zkUtils.getPartitionsForTopics(topics).apply(topic).size();
     zkUtils.close();
     return partition;
   }
 
   /**
+   * Create the topic that the monitor uses to monitor the cluster.  This method attempts to create a topic so that all
+   * the brokers in the cluster will have two partitions.
+   *
+   * @param zkUrl zookeeper connection url
+   * @param topic topic name
+   * @param topicConfig if this is null or empty or min.insync.replicas is not specified then min.insync.replicas will
+   *                    be set to 3.
+   * @return the number of partitions created
+   */
+  public static int createMonitoringTopic(String zkUrl, String topic, Properties topicConfig) {
+    ZkUtils zkUtils = ZkUtils.apply(zkUrl, ZK_SESSION_TIMEOUT, ZK_CONNECTION_TIMEOUT, JaasUtils.isZkSecurityEnabled());
+    try {
+      //TODO: throw exception? check that we have the correct number of partitions?
+      if (AdminUtils.topicExists(zkUtils, topic)) {
+        LOG.warn("Monitoring topic \"" + topic + "\" already exists.");
+        return getPartitionNumForTopic(zkUrl, topic);
+      }
+
+      int brokerCount = zkUtils.getAllBrokersInCluster().size();
+      int partitionCount = brokerCount * 2;
+
+      // see min.insync.replicas
+      if (topicConfig == null) {
+        topicConfig = new Properties();
+      }
+      if (!topicConfig.contains("min.insync.replicas")) {
+        topicConfig.setProperty("min.insync.replicas", "" + 3);
+      }
+      AdminUtils.createTopic(zkUtils, topic, partitionCount, 3 /* replication factor*/, new Properties());
```

Comment:
Thanks @smccauliff. I have commented on the latest patch. -1 is an invalid value for min.insync.replicas, right? Maybe we can simply remove this config and use replication factor - 1 as min.insync.replicas, set autoTopicCreationEnabled to true and set autoTopicReplicationFactor to 1 (just in case there is only one broker in the cluster).